example_test.go

```diff
@@ -62,6 +64,54 @@ func Example() {
 	}
 }
 
+// This example to test event broadcaster to set qps and burst size.
+func ExampleEventBroadcaster() {
+	var examplePod = &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "example",
+			Namespace: "default",
+		},
+	}
+	var log = ctrl.Log.WithName("event-examples")
+	// Controller have two diff recorderProvider
+	// 1.manager will create recorderProvider to resourceLock and push lease event.
+	// 2.cluster also create recorderProvider,when call manager.GetEventRecorderFor,will get the provider.
+	// When use the provider to push event like eventf(),provider will call getBroadcaster and StartEventWatcher.
+	// If use the EventBroadcaster options,the diff recorderProvider's makeBroadcaster func will return same broadcaster point
+	// Then call broadcaster.StartEventWatcher func duplicate,event will dunplicate push.
+	manager, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		LeaderElectionID:        "123456qbcdqe",
+		LeaderElection:          true,
+		LeaderElectionNamespace: "test1",
+		EventBroadcaster: record.NewBroadcasterWithCorrelatorOptions(record.CorrelatorOptions{
+			BurstSize: 50,
+			QPS:       15,
+		}),
+	})
+	if err != nil {
+		log.Error(err, "could not create manager")
+		os.Exit(1)
+	}
+	go func() {
+		for {
+			select {
+			case <-ctrl.SetupSignalHandler().Done():
+				return
+			case <-time.After(time.Second * 10):
+				er := manager.GetEventRecorderFor("event-test")
+				// Like this,will push the same event two times.
+				er.Eventf(examplePod, corev1.EventTypeNormal, "event-test", "EventBroadcaster test")
+				log.V(1).Info("eventf end")
+			}
+		}
+	}()
+	if err := manager.Start(ctrl.SetupSignalHandler()); err != nil {
```

Comment:
This example seems to panic because ctrl.SetupSignalHandler() is called twice?

Reply:
> This example seems to panic because ctrl.SetupSignalHandler() is called twice?

This example to test event broadcaster to set qps and burst size.
panic test is in pkg/manager/manager_test.go.

Reply:
@niuyueyang1996 you are calling SetupSignalHandler again on line 98